ovh/resource_ip_move.go

```diff
@@ -0,0 +1,309 @@
+			Computed: true,
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Description: "Possible values for ip type",
+			Computed:    true,
+		},
+		"task_status": {
+			Type:        schema.TypeString,
+			Description: "Status field of the current IP task that is in charge of changing the service the IP is attached to",
+			Computed:    true,
+		},
+		"task_start_date": {
+			Type:        schema.TypeString,
+			Description: "Starting date and time field of the current IP task that is in charge of changing the service the IP is attached to",
+			Computed:    true,
+		},
+	}
+
+	return schema
+}
+
+func resoursIpMoveCreate(d *schema.ResourceData, meta interface{}) error {
+	// later on this ID will be replaced by the task if when we need to create it (see resourceIpMoveUpdate)
+	d.SetId(d.Get("ip").(string))
+	return resourceIpMoveUpdate(d, meta)
+}
+
+// resourceIpMoveUpdate will move an ip to a provided service name or detach (= park) it otherwise
+// if the resource ID is a taskId and the previous task is not done, wait for it to be finished until the previous task is considered expired
+// (see recursiveWaitTaskFinished) before trying to do the service move.
+// then do the move only if the current service is different from the one given in the inputs
+func resourceIpMoveUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ip := d.Get("ip").(string)
+
+	// no need to update if ip is already routed to the appropriate service
+	opts, err := (&IpMoveOpts{}).FromResource(d)
+	if err != nil {
+		return err
+	}
+
+	serviceName, err := helpers.ServiceNameFromIpBlock(ip)
+	if err != nil {
+		return err
+	}
+	err = d.Set("service_name", serviceName)
+	if err != nil {
+		return err
+	}
+
+	ipTask := &IpTask{}
+	taskId, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err == nil {
+		// if previous task is not done yet we need to wait for it to be completed
+		if d.Get("task_status") != nil {
+			ipTask.Status = IpTaskStatusEnum(d.Get("task_status").(string))
+			ipTask.TaskId = taskId
+			taskStartDate, err := time.Parse("2006-01-02 15:04:05 -0700 MST", d.Get("task_start_date").(string))
+			if err != nil {
+				return err
+			}
+			ipTask.StartDate = taskStartDate
+			_, err = waitForTaskFinished(d, meta, ipTask, ip, opts)
+			if err != nil {
+				log.Printf("[WARNING] - waitForTaskFinished on previously registered task return error %s. Will continue nevertheless", err)
+			}
+		}
+	} else {
+		log.Printf("[WARNING] - resource ID %s is not an int64/not a task ID. Cannot get last task state", d.Id())
+	}
+	err = resourceIpServiceReadByServiceName(d, *serviceName, config)
+	if err != nil {
+		return err
+	}
+
+	currentlyRoutedService := GetRoutedToServiceName(d)
+	if reflect.DeepEqual(currentlyRoutedService, opts.To) {
+		log.Printf("[DEBUG] Won't do anything as ip %s (service name = %s) is already routed to service %v", ip, *serviceName, currentlyRoutedService)
+		return nil
+	} else {
+		if opts.To == nil {
```

Comment:
shouldn't we use the delete function for this ? When reading the resource, I interpret it like:
- this is a resource to associate an IP with a given service
- if you `terraform destroy` this resource, the IP is parked

Reply:
it made more sense IMO to park by defining `service_name = ""` rather than using the delete function because delete, in the terraform sense of the term, deletes a resource that was created or imported.
IP moves do not result in resource creation in the proper sense of the way : it is more like an action that results in an `ovh_ip_service` being attached to a different service, or parked, back and forth